src/Common/tests/Tests/System/Text/ValueStringBuilderTests.cs

```diff
@@ -152,6 +163,26 @@ public void Insert_IntCharInt_MatchesStringBuilder()
             Assert.Equal(sb.ToString(), vsb.ToString());
         }
 
+        [Fact]
+        public void AsSpan_ReturnsCorrectValue_DoesntClearBuilder()
+        {
+            var sb = new StringBuilder();
+            var vsb = new ValueStringBuilder();
+
+            for (int i = 1; i <= 100; i++)
+            {
+                string s = i.ToString();
+                sb.Append(s);
+                vsb.Append(s);
+            }
+
+            var resultString = new string(vsb.AsSpan());
+            Assert.Equal(sb.ToString(), resultString);
```

Comment:
nit: You could do something like this instead of using the string constructor and storing in a local:

Assert.Equal(sb.ToString(), vsb.AsSpan().ToString());